packages/edit-site/src/components/site-hub/index.js

```diff
@@ -8,24 +8,35 @@ import clsx from 'clsx';
  */
 import { useSelect, useDispatch } from '@wordpress/data';
 import { Button, __experimentalHStack as HStack } from '@wordpress/components';
+import { useViewportMatch } from '@wordpress/compose';
 import { __ } from '@wordpress/i18n';
 import { store as coreStore } from '@wordpress/core-data';
 import { decodeEntities } from '@wordpress/html-entities';
-import { memo, forwardRef } from '@wordpress/element';
+import { memo, forwardRef, useContext } from '@wordpress/element';
 import { search } from '@wordpress/icons';
 import { store as commandsStore } from '@wordpress/commands';
 import { displayShortcut } from '@wordpress/keycodes';
 import { filterURLForDisplay } from '@wordpress/url';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
 
 /**
  * Internal dependencies
  */
 import { store as editSiteStore } from '../../store';
 import SiteIcon from '../site-icon';
 import { unlock } from '../../lock-unlock';
+const { useHistory } = unlock( routerPrivateApis );
+import { SidebarNavigationContext } from '../sidebar';
 
 const SiteHub = memo(
 	forwardRef( ( { isTransparent }, ref ) => {
+		const isMobileViewport = useViewportMatch( 'medium', '<' );
+		const history = useHistory();
+		const { navigate } = useContext( SidebarNavigationContext );
+		const isRoot =
```

Comment:
Is there a better way to detect we are in the root?

This relies on the `search` being empty, which it always is at the root level, as far as I've tested. If there's some pollution (a `postType=X` remanent, for example) this wouldn't work (but also: we should clean that).

The [sidebar](https://github.com/WordPress/gutenberg/blob/db284b1a17eb78e19d1eb00d93518756c8ce07b8/packages/edit-site/src/components/sidebar-navigation-screen-main/index.js#L36) uses a explicit `isRoot` property. Given this is a temporary thing for 6.6 and will be improved by the plans/direction at https://github.com/WordPress/gutenberg/issues/60847 I didn't want to modify the public API of `SiteHub`.